src/main/java/org/example/ClientSocket.java

```diff
@@ -0,0 +1,16 @@
+package org.example;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class ClientSocket {
+
+    public Socket socketCreator(ServerSocket socket) {
+        try {
+            return socket.accept();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Why are you throwing another exception here? What's the difference between these Exceptions?
 
You are already throwing an IOExecption so in the catch you would want to do something with it, not throw another exception. For example: `e.printTrace()` [see here for more info](https://stackoverflow.com/questions/2560368/what-is-the-use-of-printstacktrace-method-in-java)

Reply:
Actually ignore previous comment. as Java Docs state 
```Constructs a new runtime exception with the specified cause and a detail message of (cause==null ? null : cause.toString()) (which typically contains the class and detail message of cause). This constructor is useful for runtime exceptions that are little more than wrappers for other throwables.```
<img width="397" alt="Screenshot 2022-11-24 at 16 07 27" src="https://user-images.githubusercontent.com/69358550/203827675-216ae749-1ae7-47c3-baa4-a4570c0876d4.png">